djangocms_navigation/admin.py

```diff
@@ -474,6 +519,51 @@ def changelist_view(self, request, menu_content_id=None, extra_context=None):
 
         return super().changelist_view(request, extra_context)
 
+    def set_main_navigation_view(self, request, menu_content_id):
+        """Sets the selected navigation as the main navigation, if possible,
+        and redirects to the changelist.
+        """
+        changelist_url = get_admin_url(self.menu_content_model, "changelist")
+        menu_content = get_object_or_404(
+            self.menu_content_model._base_manager, id=menu_content_id
+        )
+        menu = menu_content.menu
+        menu_queryset = self.menu_model._base_manager.filter(
+            site=menu.site,
+            main_navigation=True,
+            menucontent__language=menu_content.language,
+        )
+        if request.method != "POST":
+            # If this isn't a POST method, it is the first interaction, therefore redirect to confirmation
+            context = dict(
+                object_id=menu_content_id,
+                set_main_url=reverse(
+                    "admin:{app}_{model}_main_navigation".format(
+                        app=self.model._meta.app_label,
+                        model=self.menu_content_model._meta.model_name,
+                    ),
+                    args=(menu_content_id,),
+                ),
+                back_url=changelist_url,
+                extra_context=menu_queryset.values_list("identifier", flat=True),
```

Comment:
Is `menu_queryset.values_list("identifier", flat=True)` simply being used to check if there is an existing main nav object, or are you using the values somewhere? Because if you dont need the values, using `menu_queryset.exists()` would do the same thing more efficiently and is more explicit IMO.

Reply:
This is used in the confirmation template, I have refactored it to make it clearer what is going on here.